### ISSUE TITLE
The UI language should be an option the user can select

### ISSUE THREAD

**Issue by smolinari** [vuejs/vue-cli#2151]
### What problem does this feature solve?
Currently (at least I couldn't find an option to change it), the language is set automatically. I live in Germany, but I'd much rather see the UI in English. I've gone up and down the docs, but couldn't find any way to change the language. 

Thanks for taking this in consideration in advance.

Scott

### What does the proposed API look like?
Not sure about the API, but the selection could be next to the button to change to the dark mode.

**Comment by codeofsumit** [comment]
just wanted to open the same issue. No one translates programming terms like `scripts` or `dependencies` here so the UI uses terms we're not even familiar with 😅 . I would also like to change the language into english.
(I'm also from Germany)

**Comment by shershen08** [comment]
@Akryum is that a right moment to pick this feature for implementation? or I need to create a separate issue for this?

**Comment by Akryum** [comment]
No, I need to implement the settings page first.

**Comment by smolinari** [comment]
@Akryum - is there currently any way to work around the language being automatically set? Something we could hack into the CTI code? :smile: 

Scott

**Comment by reneolivo** [comment]
Yes, in my case it has a mix of Spanish/English.... it's confusing.

**Comment by agusbena** [comment]
I just can not believe this, please make it happen, I am pretty sure it can not be so dificult to add a command line option to enforce english ui. hardly anybody make software development in their mother language outside english.

**Comment by Akryum** [comment]
Do you have stats?

Le dim. 2 déc. 2018 à 02:46, Agustin Benavidez <notifications@github.com> a
écrit :

**Comment by smolinari** [comment]
> Do you have stats?

There doesn't need to be. 

It's an undocumented, but known best practice that developers should program in English if they can, simply because not doing so excludes a ton of potential from the project both in uptake and in available contributors. For closed source, maybe it's not so important, but for open source it surely is.

And, it is also a known best practice for _**any**_ program/ software to allow its users to choose what language they want to use in the application. Forcing a language on them is a no-go. It's ok to offer a certain language as a "first guess" for the user, but once the user is in, she should be offered the choice. 

Scott

**Comment by katerlouis** [comment]
Especially for newcomers like me things become really frustrating when asking for help, like in #need-help channel on the discord server. Transitioning from good old jQuery into this strange world of command line, node, npm, webpack, es6, build processes etc. is confusing enough. 

Please stop making it even harder and let me use the GUI in english!

**Comment by grezniczek** [comment]
Came here by looking for a setting to see the vue ui in English.
While I am in Germany as well, my computer runs US-English Win10, my browsers are in English, heck, every single program installed is English, yet vue ui forces me to look at German translations of things that are NEVER EVER translated by German-speaking programmers. Like all above said, there needs to be an option somewhere, and easily accessible, and make English the default and let users hunt this down if they do *not* want English.

**Comment by Akryum** [comment]
Currently it just respects the browser content language settings.

**Comment by Akryum** [comment]
Language configuration is planned as part of the upcoming Settings page/API.

**Comment by htasha** [comment]
Can somebody tell me which one is dependencies and dev dependeciens?

![image](https://user-images.githubusercontent.com/37009121/52527298-abc75b00-2c9c-11e9-9a30-fc56c2dcbe49.png)

**Comment by reneolivo** [comment]
@htasha 

Left: dependencies
Right: dev dependencies

**Comment by Globerada** [comment]
+1
Please, allow to change to english

**Comment by caugner** [comment]
Some of you have pointed out certain issues in certain locales. Please also consider contributing to the [vue-cli-locales](https://github.com/caugner/vue-cli-locales) in the meantime, by creating issues, opening merge requests or directly adding/updating strings in our [Transifex project](https://www.transifex.com/vuejs/vue-cli/). Thanks!

**Comment by smolinari** [comment]
I'm personally not going to support the translations, (like German), because 80% of it can't be translated properly anyway. It sounds like garbage. There are no German equivalents for a lot of the phrases and why I want to be able to change the language to English. 

Btw, it's like jumping through hoops to join Transifex and in the end, I never found where to go to start improving the language files. Maybe I missed it, but it certainly isn't obvious either. 

Scott

**Comment by caugner** [comment]
@smolinari You *can* change the language by setting your browser's language preference:

* Firefox: Settings > General > Language
* Chrome: Settings > Advanced > Languages

Background: https://www.w3.org/International/questions/qa-lang-priorities

**Comment by smolinari** [comment]
It's silly to have to set my browser language to get the UI to work in that language. I don't know of any app that does that. Even multi-lingual websites offer a language selection 99% of the time.

Scott

**Comment by caugner** [comment]
You *don't* have to set your browser language, you *just* have to set your browser's language *preference*.

See also: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language

Scott, while you and I may be privileged enough to understand English fluently, there are other people on this planet who don't. That's what this feature is about (in my opinion) and why it is so important.

That said, if you don't want to contribute to the locale and don't want to change your browser's language *preference*, but only want to use a locale switcher as soon as possible, I recommend you [becoming a patron](https://www.patreon.com/akryum).

PS: Scott, please reconsider if using words like "garbage" and "silly" reflect the necessary respect you would like to show towards people who create features like the UI or people who just want to help you. Thanks.

**Comment by smolinari** [comment]
> That's what this feature is about (in my opinion) and why it is so important.

I understand fully. And I'm not suggestion that feature be replaced. 

However, having a language selection in the app is for those who _want_ to see things in a different language than their browser preference and is a basic and standard app functionality too.  It _IS_ silly to suggest otherwise, which you have done.

I also have huge respect for those who do translations, as I've done them in a software myself for years (and _sold_ the German version too). I know the complications and the effort. And thus, I think I have a clue about what I am talking about, both in terms of the translation quality and the missing additional feature of language selection. 

Sorry, if my _language_ (haha, pun intended) was a bit harsh and if someone got bent out of shape because of it. It's nothing but the truth though and I don't think it's over any lines of etiquette. It just means the German translations need more work and the language selection feature is needed. 

I just reread this thread and remembered @Akryum said the language selection will be coming in a newer version. That's all I needed to know (remember). 👍 

Scott

**Comment by jbruni** [comment]
I landed in this GitHub issue page after searching for "vue ui language" in Google, right after running `vue ui` for the very first time, and :scream: when seeing the contents in Portuguese.

Time to learn how to change Chrome's language preference...

**Comment by grezniczek** [comment]
No, @jbruni , not time for you to learn how to do that, but for the developers to realize the sheer stupidity of their design decision: forcing people to make a GLOBAL change just to get one thing in the desired language, which I would guess is English for  99+% of all who actually use their product. Sigh.

**Comment by Akryum** [comment]
I'm going to lock this thread since it seems some people are not able to discuss in a respectful and civil manner. Chiming in just to post that kind of comments just discourage me to work on this more than anything.
